File: cf_276div2D.cpp
#include <stdio.h>
#include <stdlib.h>
#include <algorithm>
#include <vector>
#include <cstring>

using namespace std;

int main()
{
	int n, i,j;
	int temp;
	vector<int> bil;
	int retrievalArr[2000002];
	vector<int> unique;

	scanf("%d", &n);

	for (i=0; i<n; i++)
	{
		scanf("%d", &temp);
		bil.push_back(temp);
	}

	sort(bil.begin(), bil.end());

	memset(retrievalArr, -1, sizeof(retrievalArr));

	int prevMax = 2000001;
	for (i=n-1; i>=0; i--)
	{
		if (i!=n-1)
		{
			if (bil[i] == bil[i+1]) continue;
		}
		
		for (j=bil[i]+1; j<=prevMax; j++)
		{
			retrievalArr[j] = bil[i];
		}

		prevMax = bil[i];
		unique.push_back(bil[i]);
		
	}

	int res = 0;
	int n2 = unique.size();
	for (i=n2-1; i>=0; i--)
	{
		if (unique[i] == 1) continue;
		int x = unique[i] *2;

		while (x <= (unique[0] *2))
		{
			int temp = retrievalArr[x];
			if (temp != -1)
			{
				if (temp % unique[i] > res) res = temp%unique[i];
			}

			x+= unique[i];
		}
	}

	printf("%d\n", res);

	return 0;
}

File: cf_276div2D.py
n = input()
bil = [int(x) for x in raw_input().split(' ')]

bil.sort()

retrievalArr = [-1 for x in range(bil[len(bil)-1] *2 +2)]
maxTres = len(retrievalArr)-1
i = len(bil)-1

while (i>=0):
	if (maxTres >= bil[i]):
		retrievalArr[maxTres] = i
		maxTres-=1
	else:
		i-=1

res = 0
for j in range(len(bil)):
	x = bil[j]*2

	while (x<= (bil[len(bil)-1] *2)):
		temp = retrievalArr[x]

		if (temp != -1):
			res = max(bil[temp] %bil[j], res)

		# print j, x, temp, temp%bil[j]
		x+=bil[j]

print res
